quic/s2n-quic-transport/src/stream/stream_container.rs

```diff
@@ -537,11 +519,7 @@ impl<S: StreamTrait> StreamContainer<S> {
         );
     }
 
-    /// Iterates over all `Stream`s which are waiting for transmission,
-    /// and executes the given function on each `Stream`
-    ///
-    /// The `stream::Controller` will be notified of streams that have been
-    /// closed to allow for further streams to be opened.
+    #[cfg(test)]
     pub fn iterate_transmission_list<F>(&mut self, controller: &mut stream::Controller, mut func: F)
```

Comment:
is there a reason we need to keep this and not just merge the two?

Reply:
There are two reasons to iterate over the transmission list currently. One is to send on each stream node in the transmission list. The other is to read the order of the transmission list (used only in testing).
Basically I now need to distinguish these two codepaths, because when we're reading over the list, we _don't_ want to increment the transmission counter. Obviously we do want that behavior when sending on the list. So the two separate codepaths are now `iterate_transmission_list`, which is only used for tests, and `send_on_transmission_list`, which is what contains the counting logic.